integration/pvtdata/pvtdata_test.go

```diff
@@ -199,29 +247,109 @@ var _ bool = Describe("PrivateData", func() {
+				Output: tempCryptoDir,
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess, network.EventuallyTimeout).Should(gexec.Exit(0))
+
+			By("copying the new msp certs for peer1.org2 to the original crypto dir")
 			org2peer1 := network.Peer("org2", "peer1")
-			network.JoinChannel("testchannel", orderer, org2peer1)
-			org2peer1.Channels = append(org2peer1.Channels, &nwo.PeerChannel{Name: "testchannel", Anchor: false})
+			oldPeerMSPPath := filepath.Join(network.RootDir, "crypto", "peerOrganizations", "org2.example.com", "peers", "peer1.org2.example.com", "msp")
+			tempPeerMSPPath := filepath.Join(tempCryptoDir, "peerOrganizations", "org2.example.com", "peers", "peer1.org2.example.com", "msp")
+			os.RemoveAll(oldPeerMSPPath)
+			err = exec.Command("cp", "-r", tempPeerMSPPath, oldPeerMSPPath).Run()
+			Expect(err).NotTo(HaveOccurred())
 
-			ledgerHeight := getLedgerHeight(network, network.Peer("org1", "peer0"), "testchannel")
+			By("copying the new Admin@org2.example.com user certs to the original user certs dir as Admin2@org2.example.com")
+			oldAdminUserPath := filepath.Join(network.RootDir, "crypto", "peerOrganizations", "org2.example.com", "users", "Admin2@org2.example.com")
+			tempAdminUserPath := filepath.Join(tempCryptoDir, "peerOrganizations", "org2.example.com", "users", "Admin@org2.example.com")
+			os.RemoveAll(oldAdminUserPath)
+			err = exec.Command("cp", "-r", tempAdminUserPath, oldAdminUserPath).Run()
+			Expect(err).NotTo(HaveOccurred())
+			err = os.Rename(
+				filepath.Join(oldAdminUserPath, "msp", "signcerts", "Admin@org2.example.com-cert.pem"),
+				filepath.Join(oldAdminUserPath, "msp", "signcerts", "Admin2@org2.example.com-cert.pem"),
+			)
+			Expect(err).NotTo(HaveOccurred())
 
-			By("fetch latest blocks to peer1.org2")
-			sess, err := network.PeerAdminSession(org2peer1, commands.ChannelFetch{
-				Block:      "newest",
+			By("parsing the two MSP configs")
+			oldConfig := &mspp.MSPConfig{}
+			err = proto.Unmarshal(updatedConfig.ChannelGroup.Groups["Application"].Groups["org2"].Values["MSP"].Value, oldConfig)
+			Expect(err).NotTo(HaveOccurred())
+
+			tempOrg2Path := filepath.Join(tempCryptoDir, "peerOrganizations", "org2.example.com", "msp")
+			newConfig, err := msp.GetVerifyingMspConfig(tempOrg2Path, "Org2MSP", "bccsp")
+			Expect(err).NotTo(HaveOccurred())
+			oldMspConfig := &mspp.FabricMSPConfig{}
+			newMspConfig := &mspp.FabricMSPConfig{}
+			err = proto.Unmarshal(oldConfig.Config, oldMspConfig)
+			Expect(err).NotTo(HaveOccurred())
+			err = proto.Unmarshal(newConfig.Config, newMspConfig)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("merging the two MSP configs")
+			oldMspConfig.RootCerts = append(oldMspConfig.RootCerts, newMspConfig.RootCerts...)
+			oldMspConfig.TlsRootCerts = append(oldMspConfig.TlsRootCerts, newMspConfig.TlsRootCerts...)
+			oldMspConfig.FabricNodeOus.PeerOuIdentifier.Certificate = nil
+			oldMspConfig.FabricNodeOus.ClientOuIdentifier.Certificate = nil
```

Comment:
all this huge block where you edit the MSPConfig can and should be extracted to its own function.